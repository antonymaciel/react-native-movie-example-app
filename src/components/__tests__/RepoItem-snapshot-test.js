import React from 'react';
import renderer from 'react-test-renderer';

import RepoItem from '../componentExample';

const repo =  {
    description: 'Useful description goes here.',
    id: 1,
    name: 'Test 1',
    stargazers_count: 50000,
  };


it('renders a RepoItem using Snapshots', () => {
  expect(renderer.create(
    <RepoItem
      repo={repo}
      selectRepo={jest.fn()}
    />
  )).toMatchSnapshot();
});

it('renders a selected RepoItem using Snapshots', () => {
  expect(renderer.create(
    <RepoItem
      isSelected
      repo={repo}
      selectRepo={jest.fn()}
    />
  )).toMatchSnapshot();
});
