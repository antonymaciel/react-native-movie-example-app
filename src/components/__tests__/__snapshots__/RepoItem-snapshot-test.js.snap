// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a RepoItem using Snapshots 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Object {
        "paddingHorizontal": 24,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
          "paddingBottom": 4,
        }
      }
    >
      Test 1
    </Text>
    <Text
      style={
        Object {
          "paddingBottom": 8,
        }
      }
    >
      50000 stars
    </Text>
  </View>
</View>
`;

exports[`renders a selected RepoItem using Snapshots 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Array [
        Object {
          "paddingHorizontal": 24,
          "paddingVertical": 8,
        },
        Object {
          "backgroundColor": "#B2DFDB",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
          "paddingBottom": 4,
        }
      }
    >
      Test 1
    </Text>
    <Text
      style={
        Object {
          "paddingBottom": 8,
        }
      }
    >
      50000 stars
    </Text>
    <Text>
      Useful description goes here.
    </Text>
  </View>
</View>
`;
